lib/google_apis/generator.ex

```diff
@@ -14,15 +14,21 @@
 
 defmodule GoogleApis.Generator do
   @callback generate_client(GoogleApis.ApiConfig.t()) :: any()
-  alias GoogleApis.ApiConfig
+  alias GoogleApis.{ApiConfig, ChangeAnalyzer}
 
   require Logger
 
   def bump_version(api_config) do
-    version = api_config |> current_hex_version() |> bump_version_string()
-    set_mix_version(api_config, version)
-    requirement = version_requirement_string(version)
-    set_readme_version(api_config, requirement)
+    hex_version = current_hex_version(api_config)
+    bump_type = if ChangeAnalyzer.has_semantic_changes?(api_config), do: :minor, else: :patch
```

Comment:
This will still bump a minor if I swap 2 lines in an implementation because the AST is different although the surface area is unchanged. Should we call this function something else? Is it worthwhile to have this only consider function definitions/public surface area? I'm not sure if that's possible given any metaprogramming that may happen in the files?

Reply:
I guess we could call the function something else. Maybe `has_significant_changes?`. This isn't meant to be anything fancy. It's just there to detect a couple of very common cases (e.g. they reformat some of the comments) that we can map to patch versions.